File: javascript/day1.js
// increasing depth
const fs = require('fs')
const data = fs.readFileSync('../inputs/day1_mike.txt', {encoding: 'utf8', flag: 'r'})
const array = data.split('\n').map(num => parseInt(num))

function increasingDepth(depths) {
    let count = 0
    for (let i = 1; i < depths.length; i++) {
        if (depths[i-1] < depths[i]) {
            count+=1
        }
    }
    return count
}

console.log(increasingDepth(array)) 


function increasingDepthWindow(depths) {
    let count = 0
    for (let i = 1; i < depths.length-2; i++) {
        sumPrevious = depths[i-1] + depths[i] + depths[i+1]
        sumCurrent = depths[i] + depths[i+1] + depths[i+2]
        if (sumCurrent > sumPrevious) {
            count+=1
        }
    }
    return count
}

console.log(increasingDepthWindow(array)) 

File: python/day1.py
with open("../inputs/day1.txt", "r") as f:
    data = f.readlines()

data = [int(x) for x in data]

# part one
count = 0
for x, y in zip(data[:-1], data[1:]):
    if y > x:
        count += 1

print(f"part 1: {count}")

# part two
count = 0
for i in range(3, len(data)):
    x = sum(data[i - 3 : i])
    y = sum(data[i - 2 : i + 1])
    if y > x:
        count += 1

print(f"part 2: {count}")
